factory.py

```diff
@@ -703,6 +704,10 @@ def set_up_logging(quiet, verbose):
     logging.basicConfig(level=log_level, format='%(message)s')
 
 
+def handle_sigterm():
+    signal.signal(signal.SIGTERM, lambda *args: sys.exit(1))
```

Comment:
print / log something so we know it's our handler that gets hit, future debugging will be easier